### ISSUE TITLE
Possibility to preserve config file between different versions of software

### ISSUE THREAD

**Issue by Jamesits** [AOSC-Archive/auch#2]
Some programs fail when their config files failed to be parsed. So maintain one config file across all versions (while taking the risk of falling on early start) or mount `/etc/some_package` on the fly (which means you need to config every version of every package manually)?

**Comment by MingcongBai** [comment]
I think the original idea (not in the documentation at the moment) is that they are not managed as part of the virtrootfs. While packagers are supposed to be the ones responsible for a reasonably designed /etc hierarchy.

But yes, I think the risk of failing due to configuration flaw is still possible under this model.

**Comment by m13253** [comment]
Configurations are not isolated. Every versions of the same software share the same configuration.

Reasons:
1. Most programs maintain the same configuration format across versions. Web administrators do not expect to modify configuration files each time nginx is updated.
2. Most programs can specify the location of configuration file, probably via `--config` or `-c` options. The user can maintain them manually.

**Comment by Jamesits** [comment]
> Most programs can specify the location of configuration file, probably via --config or -c options. 

When this add up to #1 now we really need an namespace: `from vim:7.4 import vim alias "vim -c $HOME/.vim74/.vimrc"`

**Comment by m13253** [comment]
> When this add up to #1 now we really need an namespace: from vim:7.4 import vim alias "vim -c $HOME/.vim74/.vimrc"

That's okay. But auch will not do this automatically.

**Comment by Jamesits** [comment]
No, you need to enforce these aliases work across programs since you are already redirecting calls to creating process.

**Comment by m13253** [comment]
> No, you need to enforce these aliases work across programs since you are already redirecting calls to creating process.

See the reason 1 I pointed. No one want to copy and paste after upgrading vim.

**Comment by bobcao3** [comment]
I support to isolate configuration.
Just make /var/auch/pkg/etc
Nothing hurt

Also we can setting the capability reuse configs

**Comment by bobcao3** [comment]
Maybe I will need different version of config of X11 to deal with my i915 and my touchpad

**Comment by Jamesits** [comment]
@bobcao3 This can be done via a "default" config and "specific" configs which will override default if exists.

**Comment by m13253** [comment]
I have read this article: http://brandon.invergo.net/news/2012-05-26-using-gnu-stow-to-manage-your-dotfiles.html

Since GNU stow can manage this pretty well. I think it is not so necessary that we integrate this into Auch.

**Comment by Jamesits** [comment]
@m13253 stow requires configuring specific installation prefix one for each version and links them to correct folder once you need it. Seems a lot like our virtual file system implemented in another way. However it doesn't work for any packages, especially these installed by package manager. Also it requires one configure file per version, and in most cases a default config just works.
